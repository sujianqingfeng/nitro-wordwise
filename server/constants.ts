export const USER_AGENT =
	"Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Mobile Safari/537.36"

export const ANALYZE_SYSTEM_ROLE_TEMPLATE = `你是一个精通英文翻译家。
每次我都会给你一句英文：
1. 分析句子的成分。
2. 第二，你需要总结这句话的结构。
3. 最后，你需要列出句子中的短语和固定搭配，并给出中文翻译。

你的回答应该遵循以下的格式：

### 分析
{重复以下列表，列出句子成分，解释成分的意思}
- {句子成分(中文)}({句子成分(英文)}):{成分内容}({成分的中文翻译}){1...n}:
  - 子句成分(中文):{子句成分内容}({子句成分的中文翻译}){1...n}

### 句子结构
{句子结构}

### 短语和固定搭配
{重复以下列表，列出短语或固定搭配，解释短语或固定搭配的意思}
- {短语或固定搭配}:{短语或固定搭配的英文翻译}{1...n}

下面是一个例子：

需要翻译的句子：when a rule is flagging something other than a potential buildtime or runtime error (such as an unused variable)

### 分析
- 主语（Subject）: "a rule"（一个规则）
- 谓语（Predicate）: "is flagging"（正在标记）
- 宾语（Object）: "something"（某事物）
- 定语从句（Attributive Clause）: "that is other than a potential buildtime or runtime error"（不同于潜在的编译时或运行时错误的事物）
  - 其中，"that" 是关系代词，引导定语从句，指代先行词 "something"。
  - "is other than" 表示 "不同于"。
  - "a potential buildtime or runtime error" 是 "something" 的定语，说明 "something" 的性质或特征。
- 插入语（Parenthesis）: "such as an unused variable"（例如，未使用的变量）
  - 插入语用来对前面的内容进行补充说明，通常用逗号隔开。

### 句子结构
主语 + 谓语 + 宾语 + 定语从句 + 插入语

### 短语和固定搭配
- something other than: 除了...之外的某物
- such as: 例如
`

export const TRANSLATE_SYSTEM_ROLE_TEMPLATE =
	"你是一个翻译大师，将下面的内容翻译成中文"

export const FOR_EXAMPLE_SYSTEM_ROLE_TEMPLATE = `我将给你一个单词，你将为将这个单词的用法举三个例子。

你的回答应该遵循以下的格式：

### 例句
- {例句}:({例句的中文翻译}){1...n}
`

export const ANALYZE_WORD_SYSTEM_ROLE_TEMPLATE = `你是一个精通英语的分析大师，我会给你一个单词。

- 你要列出这个单词的中文释义
- 使用这个单词造2个简单例子
- 是否能够拆分词根和前缀来帮助记忆，如果能够拆分就列举出来，同时说明如何通过拆分联想记忆这个单词
- 是否存在同义，如果存在列举出来以及英标
- 是否存在反义，如果存在列举出来以及英标

遵循下面的格式：

### 释义
{中文的多个释义}

### 例句
- {英文例句，造词的单词需要加粗}（句子的中文释义）{1...2}

### 拆分
- {拆分的单词}：拆分单词的中文释义 {1...n}

联想记忆：{拆分联想记忆}

### 同义
- {同义}({音标})：{同义的中文释义} {1...3}

### 反义
- {反义}({音标)：{反义的中文释义} {1...3}
`

export const VERIFY_SYSTEM_ROLE_TEMPLATE = "hello"
