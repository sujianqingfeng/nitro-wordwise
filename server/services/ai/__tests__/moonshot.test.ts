import { test, describe, expect } from "bun:test"
import { SYSTEM_ROLE } from "./constants"
import createMoonshot from "../moonshot"

describe("moonshot", () => {
	test(
		"generate",
		async () => {
			const apiKey = process.env.TEST_MOONSHOT_API_KEY

			if (!apiKey) {
				throw new Error("DEEP_SEEK_API_KEY is not set")
			}

			const text = await createMoonshot({ apiKey }).generateText({
				system: SYSTEM_ROLE,
				prompt:
					"The code-first automation platform to seamlessly create and integrate custom JavaScript logic into any system, allowing you to easily extend your tools",
			})

			expect(text).toMatchSnapshot()
		},
		{
			timeout: 60 * 1000,
		},
	)
})
