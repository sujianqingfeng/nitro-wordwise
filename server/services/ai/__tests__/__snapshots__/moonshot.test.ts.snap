// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`moonshot generate 1`] = `
"### 分析
- code-first:
  - 词组：code-first
  - 释义：以代码为中心的开发方式，即先编写代码，再根据代码生成其他内容（如数据库模式、API文档等）。

- automation platform:
  - 词组：automation platform
  - 释义：自动化平台，指的是一个可以自动化执行任务的软件系统。

- seamlessly:
  - 词组：seamlessly
  - 释义：无缝地，指在两个或多个部分之间没有明显的界限或中断。

- create and integrate:
  - 词组：create and integrate
  - 释义：创建并集成，指开发新的内容或功能，并将其整合到现有的系统中。

- custom JavaScript logic:
  - 词组：custom JavaScript logic
  - 释义：自定义的JavaScript逻辑，指根据特定需求编写的JavaScript代码。

- any system:
  - 词组：any system
  - 释义：任何系统，指不限制于特定类型的系统或平台。

- easily extend:
  - 词组：easily extend
  - 释义：容易扩展，指可以方便地增加新功能或修改现有功能。

### 译文初稿
这是一个以代码为中心的自动化平台，可以无缝地创建并集成自定义的JavaScript逻辑到任何系统中，让你轻松扩展你的工具。

### 校对
- 校对意见1:
  - 原文：code-first
  - 译文：以代码为中心的
  - 问题：翻译较为生硬，没有很好地传达“以代码为中心”的开发方式的特点。
  - 建议：可以改为“以代码驱动的”，以更自然地表达“code-first”的含义。

- 校对意见2:
  - 原文：easily extend
  - 译文：让你轻松扩展你的工具
  - 问题：这里的“工具”可能不够准确，原文中的“tools”可能指的是更广泛的应用或系统。
  - 建议：可以改为“让你轻松扩展你的应用或系统”。

### 译文终稿
这是一个以代码驱动的自动化平台，可以无缝地创建并集成自定义的JavaScript逻辑到任何系统中，让你轻松扩展你的应用或系统。"
`;
